clients/src/main/java/org/apache/kafka/common/requests/IncrementalAlterConfigsRequest.java

```diff
@@ -106,4 +107,35 @@ public AbstractResponse getErrorResponse(final int throttleTimeMs, final Throwab
         }
         return new IncrementalAlterConfigsResponse(response);
     }
+
+    // It is not safe to print all config values
+    @Override
+    public String toString() {
```

Comment:
Can we do the "copy the whole object, then replace sensitive fields with [REDACTED]" approach here?

If we do this by hand, we'll probably forget to update it when someone adds a new field.

Plus, this will not be efficient anyway, nor does it need to be.

Reply:
got it, I see your point about forgetting to update the toString. I had thought this approach would be a bit more transparent than doing find-replace on a very generic field name ('value') which may appear in the string multiple times. 

this approach should work well for `AlterUserScramCredentialsRequest` since `salt` and `saltedPassword` are pretty unique terms.

Reply:
or are you suggesting we only redact config values for known sensitive config names? we would have to maintain a list of all sensitive config names then which seems more error prone than just removing/redacting all config values.

Reply:
thanks for the offline discussion, using the JsonConverter logic seems to simplify the logic a bunch.